### ISSUE TITLE
knife winrm commands running serially instead of in parallel with 1.4.0 after upgrading from 0.8.4

### ISSUE THREAD

**Issue by laynes** [chef/knife-windows#364]
We recently upgraded our knife-windows version in a Gemfile for our deployments and instead of executing the command in parallel on a subset of nodes, the commands all executed sequentially.  
We have anywhere from 10-30 nodes per environment, so you can imagine our deployments that used to take 10 minutes were now taking over an hour per environment.

When we rolled back the version of knife-windows version back to 0.8.4, the commands were executed in parallel again.  

Is this expected behavior?  Seems like a regression to me.  

Here's an example using ping.  As you can see from the output, when executed with 0.8.4 (the first execution with bundle exec), the execution is occurring in parallel, while with the new version 1.4.0, the execution is serial.

```
$ bundle exec knife winrm 'chef_environment:*** AND role:***' 'ping 127.0.0.1 -n 10' -x username -P '****' -c ~/.chef/knife.rb 
318240.domain.com 
318240.domain.com Pinging 127.0.0.1 
318240.domain.com with 32 bytes of data:
318240.domain.com Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
407123.domain.com 
407123.domain.com Pinging 127.0.0.1 
407123.domain.com with 32 bytes of data:
407123.domain.com Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com 
318240.domain.com Ping statistics for 127.0.0.1:
318240.domain.com     Packets: Sent = 10, Received = 10, Lost = 0 (0% loss),
318240.domain.com Approximate round trip times in milli-seconds:
318240.domain.com     Minimum = 0ms, Maximum = 0ms, Average = 0ms
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com 
407123.domain.com Ping statistics for 127.0.0.1:
407123.domain.com     Packets: Sent = 10, Received = 10, Lost = 0 (0% loss),
407123.domain.com Approximate round trip times in milli-seconds:
407123.domain.com     Minimum = 0ms, Maximum = 0ms, Average = 0ms

$ knife winrm 'chef_environment:*** AND role:***' 'ping 127.0.0.1 -n 10' -x username -P ‘****/.chef/knife.rb 
407123.domain.com 
407123.domain.com Pinging 127.0.0.1 
407123.domain.com with 32 bytes of data:
407123.domain.com Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com Reply from 127.0.0.1: 
407123.domain.com bytes=32 time<1ms TTL=128
407123.domain.com 
407123.domain.com Ping statistics for 127.0.0.1:
407123.domain.com     Packets: Sent = 10, Received = 10, Lost = 0 (0% loss),
407123.domain.com Approximate round trip times in milli-seconds:
407123.domain.com     Minimum = 0ms, Maximum = 0ms, Average = 0ms
318240.domain.com 
318240.domain.com Pinging 127.0.0.1 
318240.domain.com with 32 bytes of data:
318240.domain.com Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com Reply from 127.0.0.1: 
318240.domain.com bytes=32 time<1ms TTL=128
318240.domain.com 
318240.domain.com Ping statistics for 127.0.0.1:
318240.domain.com     Packets: Sent = 10, Received = 10, Lost = 0 (0% loss),
318240.domain.com Approximate round trip times in milli-seconds:
318240.domain.com     Minimum = 0ms, Maximum = 0ms, Average = 0ms
```

**Comment by mwrock** [comment]
closing as duplicate of #348
